graphql-dgs-codegen-shared-core/src/test/kotlin/com/netflix/graphql/dgs/client/codegen/ProjectionSerializerTest.kt

```diff
@@ -102,4 +102,37 @@ internal class ProjectionSerializerTest {
             """.trimMargin()
         )
     }
+
+    @Test
+    fun `Projection supports aliases`() {
+        // given
+        val root = EntitiesProjectionRoot()
+            .onMovie(Optional.empty())
+            .moveId().title().releaseYear()
+            .alias("colesReviews") { it.reviews("Cole", 10).username().score() }
```

Comment:
⚠️ Possible risk of this API:

In the snippet above you see how an alias is initialized, then further fields are projected (`reviews`, `username`, `score`). 

Users may also decide to write operations that have other side effects.

⚠️ Alias within an alias
```java
alias("root") { it.alias("subalias", { /* .. */ } })
```
⚠️ .parent() and .root()
```java
alias("title") { it.movieTitle().parent().root() })
```
These kinds of calls could be easily mistaken and result in unpredictable query serialization.

Is this the right API? If we should look at alternative ideas, we could gate the calling pattern by introducing a new interface whether through a proxy or further boilerplate code generation.

Reply:
Also another risk is when `alias` is a filed defined in the schema, right?

Reply:
Yes, especially when `alias` takes in an argument of a String type.